### PR TITLE
Also fail on broken -x 123, should be -x123 for now

### PR DESCRIPTION
**Pull request by benjaoming**
Fix for #4058

### CODE REVIEW

**Comment by MCGallaspy** [comment]
I want to settle this option parsing stuff once and for all, so I made a big ol' regex that should match any valid long or short form argument. Here it is:
```re
(-[0-9a-zA-Z]\s?(?P<sh_value>\w+|'[^']*')|--[\w-]+[\s=]{1}(?P<lg_value>\w+|'[^']*'))
```
And at [pythex.org, with examples](http://pythex.org/?regex=%5E(-%5B0-9a-zA-Z%5D%5Cs%3F(%3FP%3Csh_value%3E%5Cw%2B%7C%27%5B%5E%27%5D*%27)%7C--%5B%5Cw-%5D%2B%5B%5Cs%3D%5D%7B1%7D(%3FP%3Clg_value%3E%5Cw%2B%7C%27%5B%5E%27%5D*%27))&test_string=--blah-di-da%3D1correct%0A--blah%202correct%0A--blah-di-da%203correct%0A--blah%3D4correct%0A--another-long-thing%20%275correct%20quotes%20with%20spaces%27%0A--an--even-longer--long-thing%3D%276correct%20quotes%20with%20spaces%20and%20-s%27%0A-x%20123%0A-x123%0A-l%27pt-Br%27%0A-l%20%27pt-br%27%0A-b%3Ddoesntwork&ignorecase=0&multiline=1&dotall=0&verbose=0). Disregard the initial `^` there, that's so it works in multiline mode.

What this matches in plain english:

First, it either tries to match short-form or long-form arguments.

A short-form argument is matched if there is exactly 1 `-` followed by 1 alphanumeric character (`\w`). Then there is exactly one optional whitespace (in particular, = is not a valid character). Then it can match a value, as described below.

A long-form argument is matched if there are exactly two dashes (`--`) followed by 1 or more alphanumeric characters, a dash (`-`), or an underscore (`_`). This is followed by exactly 1 of a whitespace character or `=`, and then a value as described below.

Values are matched the same way for long- or short-form arguments. They can either be 1 or more alphanumeric or `_` (`\w+`) or _anything_ inside of two `'` (aside from `'` itself), including nothing. The `'` are also matched, so if you don't want 'em then you gotta strip 'em. This allows us to pass whitespace characters within arguments. This also shouldn't break any existing uses, as it continues to match them.

Do with this as you will!

**Comment by MCGallaspy** [comment]
Okay, after further discussion it's not a simple matter of dropping in a such a regex... Seems like the best course of action is to merge this in for now. However, we also decided it was better to live with the "lesser evil" of having options out of order than having to maintain a separate version of docopt. It's too much to revert it for 0.14, but let's put a moratorium on further changes and revert it for 0.15.

**Comment by benjaoming** [comment]
@MCGallaspy #4039

**Comment by benjaoming** [comment]
I've opened an upstream PR for docopt to fix some of this.
